tests/OpportunityContextUtils.test.ts

```diff
@@ -223,4 +223,176 @@ describe('OpportunityContextUtils', () => {
       });
     });
   });
+
+  describe('external provider properties', () => {
+    describe('initOpportunityContext', () => {
+      it('external created at context will be set', () => {
+        initOpportunityContext(ctx, {
+          key: OpportunityContextKeys.EXTERNAL_CREATED_AT,
+          value: '2100-01-01',
+        });
+        expect(ctx.externalCreatedAt).toStrictEqual(new Date('2100-01-01'));
+      });
+      it('external provider id context will be set', () => {
+        initOpportunityContext(ctx, {
+          key: OpportunityContextKeys.EXTERNAL_ID,
+          value: 'some-value',
+        });
+        expect(ctx.externalProviderId).toBe('some-value');
+      });
+      it('external provider name context will be set', () => {
+        initOpportunityContext(ctx, {
+          key: OpportunityContextKeys.EXTERNAL_PROVIDER,
+          value: 'some-value',
+        });
+        expect(ctx.externalProviderName).toBe('some-value');
+      });
+    });
+
+    describe('toOpportunityParams', () => {
```

Comment:
From here and below it feels way to repetitive, worth refactoring at some point in the future.